modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1404,6 +1405,87 @@ public String getName() {
         return visible;
     }
 
+    /**
+     * Whether or not this {@code Node} is showing (that is, it is part of an
+     * open Window on the user's system). The Window might be "showing", yet the user might not
+     * be able to see it due to the Window being rendered behind another Window
+     * or due to the Window being positioned off the monitor.
+     *
+     * <p>Note that the {@code Node} does not need to be visible for this property
+     * to be {@code true}.
+     *
+     * @defaultValue false
+     * @since 20
+     */
+    private ReadOnlyBooleanProperty shown;
+
+    public final boolean isShown() {
+        Scene s = getScene();
+        if (s == null) return false;
+        Window w = s.getWindow();
+        return w != null && w.isShowing();
+    }
+
+    public final ReadOnlyBooleanProperty shownProperty() {
+        if (shown == null) {
+            ObservableValue<Boolean> ov = sceneProperty()
+                .flatMap(Scene::windowProperty)
+                .flatMap(Window::showingProperty);
+
+            shown = new ReadOnlyBooleanDelegate(Node.this, "shown", ov);
+        }
+
+        return shown;
+    }
+
+    // Candidate to make publicly available or to add as a convience method to ObservableValue
+    private static class ReadOnlyBooleanDelegate extends ReadOnlyBooleanProperty {
```

Comment:
It can be, generally I tend to this when there is a second use of the class.  In that case we may also want to introduce the other primitive versions.  So for now I would leave it here, until there is another use case.